homeassistant/components/here_travel_time/sensor.py

```diff
@@ -240,6 +240,21 @@ def __init__(
         self._attr_native_unit_of_measurement = TIME_MINUTES
         self._attr_name = name
 
+    async def async_added_to_hass(self) -> None:
+        """Handle when entity is added."""
+        if self.hass.state != CoreState.running:
+            _LOGGER.info("Waiting for Home Assistant to fully start")
+            self.hass.bus.async_listen_once(
+                EVENT_HOMEASSISTANT_STARTED, self.async_first_update()
+            )
+        else:
+            await self.async_first_update()
+
+    async def async_first_update(self):
+        """Update the entity after Home Assistant has fully started."""
+        await super().async_added_to_hass()
```

Comment:
Is it safe to call this immediately in `async_added_to_hass`?  The reason I mention this is because it felt a little difficult to follow the back and forth here since this calls a parent class of the method above, and it first read like a circular dependency.   I looked at the coordinator entity update and it seemed like it could work to call it since it handles coordinator update de-registration only.

Reply:
I reread the coordinator code. I think it is the first time I fully understood it. Thanks for pointing that out!